recipes/jsoncpp/all/conanfile.py

```diff
@@ -47,6 +47,12 @@ def _patch_sources(self):
                               "add_subdirectory( example )",
                               "",
                               strict=False)
+        
+        if tools.Version(self.version) <= "0.10.7":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                                "SET(INCLUDE_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/include/jsoncpp",
+                                "SET(INCLUDE_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/include",
+                                strict=False)
```

Comment:
You should move this to a separate file and apply as a patch

Reply:
Why this patch at all? A name-prefix in include path is actually preferable to avoid naming conflicts

We also have a generic discussion about this in an issue, that I can't find right now....

Reply:
maybe #435 ?

Reply:
Exactly, #435 it is.

Anyways, since this decision to use a prefix comes here directly from upstream, I don't think we should change it

Reply:
@prince-chrismc , @Croydon 
As suggested, I let the name-prefix as it was in version "0.y.z".
Due to this, I need to add the prefix "jsoncpp" to the include in test_package for versions 0.y.z.
It results with two test_package folders. I renamed test_package to test_package_default for current versions, and I added folder test_pacakge_0.y.z for branch "0.y.z". 

What is the correct way too deal with that case? What do you suggested?

Reply:
Ahhh!

So `0.y.z` => is `#include <jsoncpp/json/json.h>`

And `1.y.z` => is `#include <json/json.h>`

---

Keeping one test package, you can use the the `"cmake_find_package"` generator, and in the CMakeLists.txt you can call `find_package(JsonCpp)` which will give you a cmake variable `JsonCpp_VERSION` which you can check to set a marco which you use in the test_package.cpp to toggle between the includes

Reply:
I updated test_package to work with versions 0.y.z

Reply:
Looks spot on!